shell/common/gl_context_switch_manager.h

```diff
@@ -0,0 +1,71 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_COMMON_GL_CONTEXT_SWITCH_MANAGER_H_
+#define FLUTTER_SHELL_COMMON_GL_CONTEXT_SWITCH_MANAGER_H_
+
+#include <map>
+#include "flutter/fml/macros.h"
+
+namespace flutter {
+
+// Manages `GLContextSwitch`.
+//
+// Should be subclassed for platforms that uses GL and requires context
+// switching. Always use `MakeCurrent` and `ResourceMakeCurrent` in the
+// `GLContextSwitchManager` to set gl contexts.
+class GLContextSwitchManager {
```

Comment:
Please elaborate the purpose of this class. Also, lets follow the doxygen guidelines for documentation. See [`shell.h`](https://github.com/flutter/engine/blob/e5692949023766c6a42b554a0ded365f1e8a33b3/shell/common/shell.h#L87) for an example.

Reply:
I want to ios_gl_context and ios_render_target_gl both access the EAGLContext in a similar way, so I created the `IOSGLContextSwitchManager`. I created this so that other platforms can follow a similar approach in the future. But not that after a second look, I might just remove this class and only leave the `IOSGLContextSwitchManager` as this PR only worries about iOS. And in future, we can further refactor if we want to support other platforms for switching. What do you think?

Reply:
done with the doxygen